x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/CreateApiKeyRequest.java

```diff
@@ -45,13 +49,19 @@ public CreateApiKeyRequest() {}
      * @param expiration to specify expiration for the API key
      */
     public CreateApiKeyRequest(String name, @Nullable List<RoleDescriptor> roleDescriptors, @Nullable TimeValue expiration) {
+        this.id = UUIDs.base64UUID();
```

Comment:
nit: I would add a comment as it is not obvious why we would generate the id ( in the same way it would be autogenerated as the doc id ) here

Reply:
Yes, I'll do that, I wanted to do that, thanks.